Libraries/Helpers/FileCleanserHelper.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using System.IO;
+
+namespace Piedone.HelpfulLibraries.Libraries.Helpers
+{
+    /// <summary>
+    /// File Cleanser Helper Class
+    /// </summary>
+    public static class FileCleanserHelper
+    {
+        /// <summary>
+        /// Gets a cleansed file name and path.
+        /// </summary>
+        /// <param name="fileNameToValidate">File Name to Validate</param>
+        /// <returns></returns>
+        public static string GetSafeFilePath(string fileNameToValidate)
+        {
+            if (string.IsNullOrEmpty(fileNameToValidate))
+            {
+                throw new ArgumentException("Parameter 'fileNameToValidate' can not be null.", nameof(fileNameToValidate));
+            }
+
+            if (fileNameToValidate.StartsWith(".") || fileNameToValidate.StartsWith("/"))
+            {
+                throw new ApplicationException($"Invalid file path '{fileNameToValidate}'.");
+            }
+
+            char[] blackListChars = Path.GetInvalidPathChars();
+
+            //Replace Black-listed Characters
+            foreach (var invalidChar in blackListChars)
+            {
+                if (fileNameToValidate.Contains(invalidChar))
+                {
+                    fileNameToValidate = fileNameToValidate.Replace(invalidChar, ' ').Trim();
```

Comment:
Why are you trimming in every iteration? Also, does it really help to change a dangerous character into a space instead of throwing an exception?

Reply:
I've moved the Trim() call down to after the character replacement.  

To answer your question, this class only cleanses the path/file name, but an additional method could be added to validate and throw an exception if needed.